src/main/java/menelaus/view/game/LevelSelectComponent.java

```diff
@@ -18,28 +18,53 @@
 import menelaus.model.LevelStars;
 import menelaus.model.SavedGames;
 
+/**
+ * The level view for the level select view.
+ * @author Obatola Seward-Evans
+ *
+ */
 public class LevelSelectComponent extends JPanel {
+	/** the level for this component. */
 	private Level level;
 	
+	/** the level stars for this level */
+	private LevelStars stars;
+	
+	/** count of all stars for this level */
+	int starCount;
+	
 	@Override
 	public Dimension getPreferredSize() {
 //		return new Dimension(80, 100);
 		return new Dimension(180, 180);
 	}
 	
 	/**
-	 * Create the panel.
+	 * Create the panel and mouse listener.
 	 */
 	public LevelSelectComponent(Level levelParam, LevelStars stars) {
 		this.level = levelParam;
+		this.stars = stars;
 		
-		setBounds(100, 100, 80, 100);
+		// Set star count
+		if ( stars == null ){
+			starCount = 0;
+		} else {
+			starCount = this.stars.getStarsCount();	
+		}
 		
 		addMouseListener(new MouseAdapter() {			
 			@Override
 			public void mouseClicked(MouseEvent e) {
-				ButtonLevelSelectController controller = new ButtonLevelSelectController(level);	
-				controller.actionPerformed(null);
+				Level nextLevel = GameWindowFrame.getInstance().getSavedGamesUtil()
+						.getNextPlayableLevelInPackage(GameWindowFrame.getInstance()
+								.getLevelsPackage());
+				
+				// decides if the level component can be clicked on or not.
+				if (starCount > 0 || level == nextLevel) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Use equals() to compare object references.](https://www.codacy.com/app/dmvouldjeff/menelaus/file/2446499320/issues/source?bid=3228196&fileBranchId=3246765#l64)